metricbeat/module/elasticsearch/shard/data_xpack.go

```diff
@@ -89,10 +90,32 @@ func eventsMappingXPack(r mb.ReporterV2, m *MetricSet, content []byte) {
 					"state_uuid":   stateData.StateID,
 				}
 				event.Index = elastic.MakeXPackMonitoringIndexName(elastic.Elasticsearch)
+				event.ID = getEventID(stateData.StateID, fields)
```

Comment:
I wonder if we really need this or if the UI keeps working if we do push new ids?

There was recently on discuss a topic that pointed out when we try to send a doc with the same id, that id actually gets rejected with an errors. Let's make sure we test this.

Reply:
Found the issue: https://github.com/elastic/beats/issues/8534

Reply:
> I wonder if we really need this or if the UI keeps working if we do push new ids?

Before this PR, we were pushing new IDs as we normally do. Unfortunately this was causing the monitoring UI to break. Specifically on the Index Listing page and the Index Detail page, the number of shards would just keep going up for an index.

That was the whole reason I had to put up this PR :(

> There was recently on discuss a topic that pointed out when we try to send a doc with the same id, that id actually gets rejected with an errors. Let's make sure we test this.

> Found the issue: #8534

I see. I tested this with default settings and it worked but let me also test it with increasing the `flush.timeout` to something like 30s to make multiple indexing/creation requests for the same event/doc in one bulk batch.

Reply:
I tested this with default settings (`queue.mem.flush.timeout: 1s`) again, and also with setting `queue.mem.flush.timeout: 30s` and later with `queue.mem.flush.timeout: 120s` to batch up multiple events into a single bulk request. In each of these cases I spun up a 2nd ES node to cause shard allocations and then later spun it down again.

I checked the docs in the `.monitoring-es-6-mb-* indices as well as the UI in each test and things looked good. The metricbeat logs are clear too (no 409s or other errors).

@ruflin Can you think of any other tests I could do?

Reply:
Then it SGTM.

Reply:
I think I'm missing something in the other issue then.

Reply:
Honestly, that other issue does seem related to this PR, but I'm unable to repro that issue with this PR.